src/lib/converter/utils/types.ts

```diff
@@ -0,0 +1,45 @@
+import * as ts from 'typescript';
+
+/**
+ * Returns the type parameters of a given type.
+ * @param type The type whos type parameters are wanted.
+ * @returns The type parameters of the type. An empty array if the type has no type parameters.
+ */
+export function getTypeParametersOfType(type: ts.Type): ts.TypeParameterDeclaration[] {
+    for (const declaration of type.symbol.declarations) {
+        if ((ts.isClassDeclaration(declaration) || ts.isInterfaceDeclaration(declaration)) &&
+             declaration.typeParameters) {
+            return declaration.typeParameters.map(tp => tp);
+        }
+    }
+
+    return [];
+}
+
+/**
+ * Returns a list of type arguments. If a type parameter has no corresponding type argument, the default type
+ * for that type parameter is used as the type argument.
+ * @param typeParams The type parameters for which the type arguments are wanted.
+ * @param typeArguments The type arguments as provided in the declaration.
+ * @returns The complete list of type arguments with possible default values if type arguments are missing.
+ */
+export function getTypeArgumentsWithDefaults(
+    typeParams: ts.TypeParameterDeclaration[],
+    typeArguments?: ts.NodeArray<ts.TypeNode>
+): ts.NodeArray<ts.TypeNode> {
```

Comment:
Why the node array? Is there some reason we can't just use `ts.TypeNode[]`?

Reply:
Unfortunately a `NodeArray` is required by `context.inherit` later on. But I changed it a bit.